ScarletEngine/include/RAL/RALResources.h

```diff
@@ -43,19 +43,49 @@ namespace ScarletEngine
 		WeakPtr<TextureHandle> AssetHandle;
 	};
 
-	class RALVertexBuffer
+	enum class RALBufferUsage
+	{
+		STATIC_DRAW,
+		STATIC_READ,
+		STATIC_COPY,
+		DYNAMIC_DRAW,
+		DYNAMIC_READ,
+		DYNAMIC_COPY,
+		INVALID
+	};
+
+	class RALGpuBuffer
 	{
 	public:
-		RALVertexBuffer(uint32_t InSize, uint32_t InUsage)
+		RALGpuBuffer(uint32_t InSize, RALBufferUsage InUsage)
 			: Size(InSize)
 			, Usage(InUsage)
 		{}
-		
+
 		virtual void UploadData(void* DataPtr, size_t Size) const = 0;
 		virtual void Release() = 0;
-	protected:
+
 		uint32_t Size;
-		uint32_t Usage;
+		RALBufferUsage Usage;
+	};
+
+	class RALVertexArray
```

Comment:
afaik VAO's are opengl specific, kind of weird to leak this into RAL. Should probably figure out a more general solution to this in the future